centreon/src/Core/ServiceTemplate/Application/UseCase/AddServiceTemplate/AddServiceTemplate.php

```diff
@@ -0,0 +1,373 @@
+     * @throws \Throwable
+     */
+    public function assertIsValidTimePeriod(?int $timePeriodId): void
+    {
+        if ($timePeriodId !== null && ! $this->timePeriodRepository->exists($timePeriodId)) {
+            $this->error('Time period does not exist', ['check_timeperiod_id' => $timePeriodId]);
+
+            throw ServiceTemplateException::idDoesNotExist('check_timeperiod_id', $timePeriodId);
+        }
+    }
+
+    /**
+     * @param int|null $iconId
+     *
+     * @throws ServiceTemplateException
+     * @throws \Throwable
+     */
+    public function assertIsValidIcon(?int $iconId): void
+    {
+        if ($iconId !== null && ! $this->imageRepository->existsOne($iconId)) {
+            $this->error('Icon does not exist', ['icon_id' => $iconId]);
+
+            throw ServiceTemplateException::idDoesNotExist('icon_id', $iconId);
+        }
+    }
+
+    /**
+     * @param int|null $notificationTimePeriodId
+     *
+     * @throws ServiceTemplateException
+     * @throws \Throwable
+     */
+    public function assertIsValidNotificationTimePeriod(?int $notificationTimePeriodId): void
+    {
+        if ($notificationTimePeriodId !== null && ! $this->timePeriodRepository->exists($notificationTimePeriodId)) {
+            $this->error(
+                'Notification time period does not exist',
+                ['notification_timeperiod_id' => $notificationTimePeriodId]
+            );
+
+            throw ServiceTemplateException::idDoesNotExist('notification_timeperiod_id', $notificationTimePeriodId);
+        }
+    }
+
+    /**
+     * @param AddServiceTemplateRequest $request
+     *
+     * @throws \Exception
+     * @throws AssertionFailedException
+     *
+     * @return NewServiceTemplate
+     */
+    private function createNewServiceTemplate(AddServiceTemplateRequest $request): NewServiceTemplate
+    {
+        $inheritanceMode = $this->optionService->findSelectedOptions(['inheritance_mode']);
+        $inheritanceMode = isset($inheritanceMode['inheritance_mode'])
+            ? $inheritanceMode['inheritance_mode']->getValue()
+            : 0;
+
+        $serviceTemplate = new NewServiceTemplate($request->name, $request->alias);
+        foreach ($request->commandArguments as $argument) {
+            $serviceTemplate->addCommandArgument($argument);
+        }
+
+        foreach ($request->eventHandlerArguments as $argument) {
+            $serviceTemplate->addEventHandlerArgument($argument);
+        }
+
+        foreach (NotificationTypeConverter::fromBits($request->notificationTypes) as $notificationType) {
+            $serviceTemplate->addNotificationType($notificationType);
+        }
+
+        $serviceTemplate->setContactAdditiveInheritance(
+            $inheritanceMode ? $request->isContactAdditiveInheritance : false
+        );
+
+        $serviceTemplate->setContactGroupAdditiveInheritance(
+            $inheritanceMode ? $request->isContactGroupAdditiveInheritance : false
```

Comment:
$inheritanceMode could be egal to '1' , '2' or '3'
```suggestion
            $inheritanceMode === '1' ? $request->isContactAdditiveInheritance : false
        );

        $serviceTemplate->setContactGroupAdditiveInheritance(
            $inheritanceMode === '1' ? $request->isContactGroupAdditiveInheritance : false
```